azure-devops/azext_devops/dev/team/security_group.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_devops.devops_sdk.exceptions import AzureDevOpsClientRequestError
+from azext_devops.devops_sdk.v5_0.graph.models import (JsonPatchOperation,
+                                                       GraphSubjectLookup,
+                                                       GraphSubjectLookupKey)
+from azext_devops.dev.common.identities import resolve_identity_as_id
+from azext_devops.dev.common.services import (get_graph_client,
+                                              get_project_id_from_name,
+                                              resolve_instance)
+from .security_group_helper import (GraphGroupVstsCreationContext,
+                                    GraphGroupMailAddressCreationContext,
+                                    GraphGroupOriginIdCreationContext)
+
+logger = get_logger(__name__)
+
+
+def list_groups(project=None, continuation_token=None, subject_types=None, organization=None, detect=None):
+    """ List all groups.
+    :param str project: List groups for a particular project.
+    :param str continuation_token : If there are more results than can't be returned in a single page, the result set
+                                    will contain a continuation token for retrieval of the next set of results.
+    :param [str] subject_types: A comma separated list of user subject subtypes to reduce the retrieved results.
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    client = get_graph_client(organization)
+    scope_descriptor = None
+    if project is not None:
+        project = get_project_id_from_name(organization, project)
+        scope_descriptor = get_descriptor_from_storage_key(project, client)
+    if subject_types is not None:
+        subject_types = subject_types.split(',')
+    group_list_response = client.list_groups(scope_descriptor=scope_descriptor,
+                                             continuation_token=continuation_token, subject_types=subject_types)
+    if group_list_response.continuation_token is not None:
+        print('Showing only 500 groups. ' +
+              'To list next set of groups use this token as --continuation-token argument and run the command again.' +
+              ' TOKEN:', group_list_response.continuation_token)
+    return group_list_response.graph_groups
+
+
+def create_group(name=None, description=None, origin_id=None, email_id=None,
+                 groups=None, project=None, organization=None, detect=None):
+    """Create a new Azure DevOps group or materialize an existing AAD group.
```

Comment:
this is debatable:
create vs materialize look like 2 different operations and maybe should have 2 different commands..

Reply:
like, can I use `groups` when materializing AAD group?

Reply:
Yes, i can use groups when materializing AAD group. I don't think adding a separate command would add much value, since functionality would be almost same as this one. 
Don't you think adding examples would be enough instead of a separate command ?
@geverghe  Thoughts?

Reply:
@gauravsaralMs - I think it's fair - it's fundamentally creating a group in Azure DevOps. IMO, the rest are implementation nitty gritty. The reasoning is as follows:
Can I create a group in Azure DevOps using cli? Yes, use group command.
Can I create a group for an AAD group in Azure DevOps? Yes, use group command.
Do you see a disconnect in the message?

Reply:
I have added few examples for create command. @gauravsaralMs  and @geverghe  Let me know if this would be useful enough to consume this command.

Reply:
not sure about users, but looked a bit confusing to me. 
I am fine with any approach

Reply:
Removing any context related to 'materializing a group' , since its internal to implementation details